File: CS/WindowsApplication1/Form1.cs
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using System;
using System.Data;
using System.Diagnostics;
using System.Windows.Forms;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        public DataTable GetData()
        {
            var dt = new DataTable();
            dt.Columns.Add("Col1", typeof(string));
            dt.Columns.Add("Col2", typeof(DateTime));
            dt.Columns.Add("Col3", typeof(Int32));
            for (int i = 0; i <= 10; i++)
                dt.Rows.Add(new object[] { i, DateTime.Today, i });
            return dt;
        }
        private void simpleButton1_Click(object sender, EventArgs e)
        {
            GridControl grid = new GridControl();
            grid.BindingContext = new BindingContext();
            GridView gridview = new GridView();
            grid.MainView = gridview;
            gridview.GridControl = grid;

            grid.DataSource = GetData();

            grid.ForceInitialize();
            gridview.PopulateColumns();

            grid.ExportToXls("1.xls");
            Process.Start("1.xls");
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports System
Imports System.Data
Imports System.Diagnostics
Imports System.Windows.Forms

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Public Function GetData() As DataTable
			Dim dt = New DataTable()
			dt.Columns.Add("Col1", GetType(String))
			dt.Columns.Add("Col2", GetType(DateTime))
			dt.Columns.Add("Col3", GetType(Int32))
			For i As Integer = 0 To 10
				dt.Rows.Add(New Object() { i, DateTime.Today, i })
			Next i
			Return dt
		End Function
		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			Dim grid As New GridControl()
			grid.BindingContext = New BindingContext()
			Dim gridview As New GridView()
			grid.MainView = gridview
			gridview.GridControl = grid

			grid.DataSource = GetData()

			grid.ForceInitialize()
			gridview.PopulateColumns()

			grid.ExportToXls("1.xls")
			Process.Start("1.xls")
		End Sub
	End Class
End Namespace
